### PR TITLE
[Test] Add warmup to the golden layer

### PR DESCRIPTION
**Pull request by zhoonit**
## Commits to be reviewed in this PR

<details><summary>[Test] Add warmup to the golden layer</summary><br />

This patch add warmup forwarding to the layer golden test

**Self evaluation:**
1. Build test: [X]Passed [ ]Failed [ ]Skipped
2. Run test: [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: Jihoon Lee <jhoon.it.lee@samsung.com>

</details>

<details><summary>[Test] Add conv2d golden tests</summary><br />

**Changes proposed in this PR:**
- Conv2d Golden tests
- remove _golden_ to the name of test cases, as it is attached in the
extension

**Self evaluation:**
1. Build test: [X]Passed [ ]Failed [ ]Skipped
2. Run test: [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: Jihoon Lee <jhoon.it.lee@samsung.com>

</details>

### CODE REVIEW

**Comment by zhoonit** [comment]
@kparichay Added warmup forwarding as discussed earlier.

File: test/input_gen/recorder.py [hunk 1]
```diff
@@ -402,6 +402,10 @@ def record_single(layer, input_shape, test_name, call_args={}):
     inputs = _rand_like(input_shape)
 
     initial_weights = [tf.Variable(i) for i in layer.weights]
+
+    for _ in range(4):
+        layer.call(inputs, **call_args) # warm layer multiple times
+
     with tf.GradientTape(persistent=True) as tape:
```
Comment (kparichay) [inline test/input_gen/recorder.py:409]: how about including this to the loop as well?

Comment (zhoonit) [reply test/input_gen/recorder.py:409]: This bit is easier to debug so let's stick with this until we have clear reason not to.

File: test/unittest/layers/layers_golden_tests.cpp [hunk 2]
```diff
@@ -227,6 +227,11 @@ TEST_P(LayerGoldenTest, run) {
   bool skip_calc_grad = shouldSkipCalcGrad();
   bool skip_calc_deriv = shouldSkipCalcDeriv();
 
+  for (int i = 0; i < 4; ++i) {
+    /// warm layer multiple times
+    layer->forwarding(rc, !shouldForwardWithInferenceMode());
+  }
+
   layer->forwarding(rc, !shouldForwardWithInferenceMode());
```
Comment (kparichay) [inline test/unittest/layers/layers_golden_tests.cpp:235]: this forwards 5 times in total but the reference code is only forwarding 4 times. I wonder how the unittests are passing?

Comment (zhoonit) [reply test/unittest/layers/layers_golden_tests.cpp:235]: test/input_gen/recorder.py:407 has extra one forward more, corresponeding with current structure

Comment (kparichay) [reply test/unittest/layers/layers_golden_tests.cpp:235]: I am not sure but test/input_gen/recorder.py:407 just seems to have a loop of 4, but in cpp code, there is a loop of 4 + an extra forward outside the loop. Please check.

Comment (zhoonit) [reply test/unittest/layers/layers_golden_tests.cpp:235]: Sorry, it's 407 in the old PR but it's 411. in the new PR

https://github.com/nnstreamer/nntrainer/pull/1589/files/ffc62cc06223bc1beed96b86deeca1e2a084893b..209f02c9972c8f72cfeed1c6804dbac9cfb1ae47#diff-5cd4cf5b490627ad8cb536f2ceaf55b7361475614f8b33873a5879dc7d0cb20dR411

Comment (kparichay) [reply test/unittest/layers/layers_golden_tests.cpp:235]: aah got it. It might be good to add some comments related to this in both py and cpp code.

**Review by kparichay — APPROVED** [review]
LGTM

File: test/input_gen/recorder.py (hunk 1 above)
Comment (kparichay) [reply test/input_gen/recorder.py:409]: cool, this can be done later.

**Pull request merged**